HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiSignatureList.java

```diff
@@ -0,0 +1,204 @@
+ * @throws IOException If there's a problem parsing the signature data.
+ */
+private void processSignatureList(final byte[] efiSigData)
+             throws CertificateException, NoSuchAlgorithmException, IOException {
+  efiSigDataIS = new ByteArrayInputStream(efiSigData);
+  while (efiSigDataIS.available() > 0) {
+    UefiSignatureData tmpSigData = new UefiSignatureData(efiSigDataIS, signatureType);
+    if (!tmpSigData.isValid()) {
+       valid = false;
+       status = tmpSigData.getStatus();
+       break;
+    }
+    sigList.add(tmpSigData);
+   numberOfItems++;
+  }
+}
+
+/**
+ * Method for processing a set of EFI SignatureList(s).
+ * @param sigData  Byte array holding one or more SignatureLists.
+ * @throws CertificateException If there's a problem parsing the X509 certificate.
+ * @throws NoSuchAlgorithmException if there's a problem hashing the certificate.
+ * @throws IOException If there's a problem parsing the signature data.
+ */
+private void processSignatureData(final ByteArrayInputStream sigDataIS)
+             throws CertificateException, NoSuchAlgorithmException, IOException {
+  while (sigDataIS.available() > 0) {
+     UefiSignatureData tmpigData = new UefiSignatureData(sigDataIS, signatureType);
+      if (!tmpigData.isValid()) {
+         valid = false;
+          status = tmpigData.getStatus();
+           break;
+       }
+         sigList.add(tmpigData);
+         numberOfItems++;
+       }
+ }
+
+/**
+ * Returns an ArrayList of EFISignatureData objects.
+ * @return ArrayList of EFISignatureData objects.
+ */
+public ArrayList<UefiSignatureData> getSigatureDataList() {
+   return sigList;
+}
+
+/**
+ * Return the number of certificates found within a certificate list.
+ * @return int Number of certs.
+ */
+public int getNumberOfCerts() {
+  return numberOfItems;
+}
+
+/**
+ * Checks to see if GUID is listed on page 1729 of UEFI spec version 2.8.
+ * @param guid GUID of the has algorithm.
+ * @return true if the GUID is a valid GUID for Signature List Type, false if not.
+ */
+public boolean isValidSigListGUID(final UefiGuid guid) {
+  switch (guid.getVendorTableReference()) {
+    case "EFI_CERT_SHA256_GUID": return true;
+    case "EFI_CERT_X509_SHA256": return true;
+    case "EFI_CERT_X509_SHA384": return true;
+    case "EFI_CERT_X509_SHA512": return true;
+    case "EFI_CERT_X509_GUID":   return true;
+    default: return false;
+  }
+}
+
+/**
+ * Provides a description of the fields within the EFI Signature Data field.
+ * Which is essentially a list of X509 certificates.
+ * @return human readable description.
+ */
+public String toString() {
+  StringBuffer sigInfo = new StringBuffer();
+  sigInfo.append("UEFI Signature List Type = " + signatureType.toString() + "\n");
+  sigInfo.append("Number if items = " + numberOfItems + "\n");
+  sigList.iterator();
+  for (int i = 0; i < sigList.size(); i++) {
+    UefiSignatureData certData = (UefiSignatureData) sigList.get(i);
```

Comment:
Create variable outside of loop structures.

Reply:
ditto